app/services/decorators/other_income_source_decorator.rb

```diff
@@ -4,16 +4,32 @@ def initialize(record)
       @record = record
     end
 
-    def as_json
-      {
-        name: @record.name,
-        monthly_income: @record.monthly_income,
-        payments: payments
-      }
+    def as_json # rubocop:disable Metrics/MethodLength
+      case @record.version
+      when CFEConstants::LATEST_ASSESSMENT_VERSION
+        {
+          monthly_equivalents: {
+            bank_transactions: find_by(transaction_type: :bank),
+            cash_transactions: find_by(transaction_type: :cash),
+            all_sources: find_by(transaction_type: :all_sources)
+          }
+        }
+      else
+        {
+          name: @record.name,
+          monthly_income: @record.monthly_income,
+          payments: payments
+        }
+      end
     end
 
     private
 
+    def find_by(transaction_type:)
+      income_categories = CFEConstants::VALID_INCOME_CATEGORIES.map(&:to_sym)
+      income_categories.reduce({}) { |hash, cat| hash.update(cat => @record["#{cat}_#{transaction_type}"]) }
+    end
+
```

Comment:
I don't like this.  First off, the method name is misleading.  `#find_by` is a common ActiveRecord method used for retrieving a record from the database, so it gives a misleading impression.  Secondly, it's not at all clear what's going on.  I'd much rather see something more verbose, but clear in intention.

Reply:
I dipped my toe in the one liner water and the intention was found wanting. I thought there was a risk 😄 . 

With nomenclature, again I meant to update this, tricky as I was thinking I'm trying to find transactions of a particular type in the record but I know that clashes with the '#find_by' ActiveRecord standard. Would something like '#record_transaction_types` be more appropriate? It looks like naming conventions for methods prefixed with a verb tend to gear towards DB interactions, and anything else is more noun orientated. Is that sort of right out of interest? Open to any suggestions too.

Reply:
Looking at it closer we are getting income transactions by type, so ended up with this
```
    def income_transactions(transaction_type:)
      income_transactions = {}

      INCOME_CATEGORIES.each do |category|
        income_transactions[category] = @record["#{category}_#{transaction_type}"]
      end

      income_transactions
    end
    ```
    let me know if that works for you

Reply:
there's also the possibility of sticking this in a concern and reusing it for disposable outgoings and having something like 
```
    def transactions(categories:, transaction_type:)
      transactions = {}

      categories.each do |category|
        transactions[category] = @record["#{category}_#{transaction_type}"]
      end

      transactions
    end
    ```
    but we can worry about that when doing the next stage maybe

Reply:
Much much better.  There is an argument for making it a concern, but then that means you have to parameterise the INCOME_CATEGORIES  / OUTGOING_CATEGORIES, so may not be worth it.  I'll leave that up to you to decide when you come to to the outogings, if you pick that ticket up.